src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/publicip/sdk/AzurePublicIPAddressOperationSDKTest.java

```diff
@@ -0,0 +1,246 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.publicip.sdk;
+
+import org.apache.log4j.Level;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.network.NetworkInterface;
+import com.microsoft.azure.management.network.NetworkSecurityGroup;
+import com.microsoft.azure.management.network.PublicIPAddress;
+import com.microsoft.azure.management.network.PublicIPAddresses;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+
+import cloud.fogbow.ras.constants.Messages;
+import cloud.fogbow.ras.core.LoggerAssert;
+import cloud.fogbow.ras.core.TestUtils;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.AzureTestUtils;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureClientCacheManager;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureGeneralUtil;
+import rx.Completable;
+import rx.Observable;
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ Azure.class, AzurePublicIPAddressSDK.class, AzureClientCacheManager.class})
+public class AzurePublicIPAddressOperationSDKTest {
+
+    private AzurePublicIPAddressOperationSDK operation;
+    private LoggerAssert loggerAssert;
+    private String resourceGroupName;
+
+    @Before
+    public void setup() {
+        // The scheduler trampoline makes the subscriptions execute in the current thread
+        Scheduler scheduler = Schedulers.trampoline();
+        this.resourceGroupName = AzureTestUtils.DEFAULT_RESOURCE_GROUP_NAME;
+        this.operation = Mockito.spy(new AzurePublicIPAddressOperationSDK(this.resourceGroupName));
+        this.operation.setScheduler(scheduler);
+        this.loggerAssert = new LoggerAssert(AzurePublicIPAddressOperationSDK.class);
+    }
+
+    // test case: When calling the subscribeAssociatePublicIPAddress method and the
+    // observable executes without any error, it must verify than returns the right
+    // logs.
+    @Test
+    public void testSubscribeAssociatePublicIPAddressSuccessfully() {
+        // set up
+        Azure azure = PowerMockito.mock(Azure.class);
+        String instanceId = AzureGeneralUtil.defineInstanceId(AzureTestUtils.RESOURCE_NAME);
+
+        NetworkInterface networkInterface = Mockito.mock(NetworkInterface.class);
+        Observable<NetworkInterface> observable = Observable.defer(() -> {
```

Comment:
You should use a method that belongs to AzureTestUtils. 
[Link](https://github.com/fogbow/resource-allocation-service/blob/develop/src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/AzureTestUtils.java#L46)

Reply:
The "Observable<Indexable>" type of the createSimpleObservableSuccess method in AzureTestUtils is not compatible with "Observable<NetworkInterface>".